pkg/aggregator/aggregator.go

```diff
@@ -154,8 +154,28 @@ func (s *referenceWalker) Start() {
 	}
 }
 
+// usedDefinitionForSpec returns a map with all used definition in the provided spec as keys and true as values.
+func usedDefinitionForSpec(sp *spec.Swagger) map[string]bool {
+	usedDefinitions := map[string]bool{}
+	walkOnAllReferences(func(ref spec.Ref) spec.Ref {
+		if refStr := ref.String(); refStr != "" {
+			if strings.HasPrefix(refStr, definitionPrefix) {
+				usedDefinitions[refStr[len(definitionPrefix):]] = true
+			}
+		}
+		return ref
+	}, sp)
+	return usedDefinitions
+}
+
 // FilterSpecByPaths remove unnecessary paths and unused definitions.
 func FilterSpecByPaths(sp *spec.Swagger, keepPathPrefixes []string) {
+	// Walk all references to find all used definitions. This function
```

Comment:
this is a strange semantics. I had to read this sentence multiple time to get it. So essentially we are garbage collecting definitions previously used by filtered-out paths? Can you write something like that in the godoc?

Reply:
Done.